vendorpromo/processors/__init__.py

```diff
@@ -1,4 +1,14 @@
-from vendorpromo.config import VENDOR_PROMO_PROCESSOR
 from django.utils.module_loading import import_string
+from django.core.exceptions import ObjectDoesNotExist
+from vendorpromo.config import PromoProcessorSiteConfig
+from siteconfigs.models import SiteConfigModel
 
-PromoProcessor = import_string('vendorpromo.processors.{}'.format(VENDOR_PROMO_PROCESSOR))
+
+def get_site_promo_processor(site):
+    site_processor = PromoProcessorSiteConfig()
```

Comment:
pass in site to site processor. same note from vendor about grabbing the instance from site_processor as well